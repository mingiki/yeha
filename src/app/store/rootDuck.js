import { all } from "redux-saga/effects";
import { combineReducers } from "redux";

import * as session from "./modules/session";
import * as auth from "./modules/auth";
import * as video from "./modules/video";
import * as main from "./modules/main";
import * as config from "./modules/config";

export const rootReducer = combineReducers({
  session : session.reducer,
  auth : auth.reducer,
  video : video.reducer,
  main : main.reducer,
  config : config.reducer
});

