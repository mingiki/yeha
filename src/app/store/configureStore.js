import { createStore } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

import {rootReducer} from './rootDuck'

const persistConfig = {
  key: 'root',
  storage, 
  whitelist: ['auth','group','instructor', 'config', 'membership' , 'lesson']
}

const persistedReducer = persistReducer(persistConfig, rootReducer)
const store = createStore(persistedReducer);

export const persistor = persistStore(store);
export default store;
