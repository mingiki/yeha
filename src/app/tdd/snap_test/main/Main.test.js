import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import store, { persistor } from "../../../store/store";


import MainComponent from '../../../components/main/MainComponent';

describe('Main', () => {
  let component = null;

  it('renders correctly', () => {
    component = renderer.create(
        <Provider store={store}>
            <PersistGate persistor={persistor}>
                <MainComponent />
            </PersistGate>
        </Provider>
        );
  });

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});