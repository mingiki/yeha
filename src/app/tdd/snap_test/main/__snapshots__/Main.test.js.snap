// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot 1`] = `null`;
